src/classes/dice/DiceRoller.ts

```diff
@@ -206,41 +206,57 @@ class DiceRoller {
     } else {
       let total = 0
       const rawRolls: number[] = []
-      const lowRolls: number[] = []
+      const rollClass: string[] = []
       let okRerolls = 0
       let staticBonus = 0
 
       staticBonus = parsedRoll.modifier
       total = staticBonus
 
       parsedRoll.dice.forEach(dieSet => {
-        if (critical) {
-          let x, y
-          x = DiceRoller.rollDieSet(dieSet, overkill)
-          y = DiceRoller.rollDieSet(dieSet, overkill)
-          if (x.result > y.result) {
-            rawRolls.push(...x.rolls)
-            lowRolls.push(...y.rolls)
-          }
-          else {
-            rawRolls.push(...y.rolls)
-            lowRolls.push(...x.rolls)
-          }
-          okRerolls += x.result > y.result ? x.rerolls : y.rerolls
-          total += x.result > y.result ? x.result : y.result
-        } else {
-          let x
-          x = DiceRoller.rollDieSet(dieSet, overkill)
-          rawRolls.push(...x.rolls)
-          okRerolls += x.rerolls
-          total += x.result
-        }
+        let x
+        x = DiceRoller.rollDieSet(dieSet, overkill, critical)
+        rawRolls.push(...x.rolls)
+        rollClass.push(...this.classifyDamageRolls(dieSet,x.rolls,overkill))
+        okRerolls += x.rerolls
+        total += x.result
       })
 
-      return new DamageRollResult(diceString, total, rawRolls, lowRolls, staticBonus, okRerolls, false)
+      return new DamageRollResult(diceString, total, rawRolls, rollClass, staticBonus, okRerolls, false)
     }
   }
 
+  public static classifyDamageRolls(dieSet: DieSet, rolls: number[], overkill?: boolean): string[] {
+    if (dieSet.quantity <= 0 || rolls.length <= 0) return []
+    
+    const rollClass: string[] = []
+    //Set initial 'overkill' and 'low' values
+    rolls.forEach((value, index) => {
+      if (overkill && (value === 1)) {
+        rollClass.push('overkill')
+      } else {
+        rollClass.push('low')
+      }
+    })
```

Comment:
I realized after pushing that this whole forEach to set the initial value of rollClass can be a much smaller roll.map() call. Would you like me to make that change?

Reply:
I think this code (and the other comments here) are perfectly fine and readable, this (and the other ideas you've marked) are totally at your discretion; I'm good either way